sycl/include/sycl/handler.hpp

```diff
@@ -381,6 +381,8 @@ class __SYCL_EXPORT handler {
                             "a single kernel or explicit memory operation.");
   }
 
+  void throwIfGraphAssociated(const std::string ExceptionMsg);
```

Comment:
I'm wondering if there is a way to eliminate the parameter from this function, and bring it more in line with the existing infrastructure to throw exceptions when something goes wrong. Since we only want to pass the info what went wrong, we could also use a templated function, similar to `throwOnLocalAccessorMisuse()`